canary.go

```diff
@@ -38,17 +40,20 @@ func main() {
 	}
 
 	for _, tunnel := range tunnels {
-		conn, err := net.DialTimeout("tcp", tunnel.Host, time.Second*10)
+		var d net.Dialer
+		ctx, _ := context.WithDeadline(context.Background(), time.Now().Add(1*time.Second))
```

Comment:
`WithTimeout` is the more natural way of doing this. You still want the defer cancel as you had before too. I'd say setting this to 1 minute per is fine, we just want to make sure it doesn't hang forever.

Reply:
ohhh the 1 second was trying something and forgot to revert that